libkbfs/md_journal.go

```diff
@@ -338,42 +398,51 @@ func (j mdJournal) convertToBranch(
 			brmd.Revision, id, newID)
 	}
 
+	// TODO: Do the below atomically on the filesystem level.
```

Comment:
You can't, but you can restructure the layout so that the final swap can be atomic, e.g. have "md_journal" always be a symlink, and then have the swap be changing it to point from one dir to another. I'll put that in a comment.